v2/assertions.go

```diff
@@ -113,7 +113,7 @@ func normalizeLF(d []byte) []byte {
 func (g *Goldie) AssertWithTemplate(t *testing.T, name string, data interface{}, actualData []byte) {
 	t.Helper()
 	if *update {
-		err := g.Update(t, name, actualData)
+		err := g.UpdateWithTemplate(t, name, data, actualData)
```

Comment:
I think this behavior is pretty nice as the default behavior. That said, I'm a little concerned about the backwards compatibility. For example, we could introduce a `--update-with-template` or so flag which would trigger the UpdateWithTemplate behavior.

Reply:
Rather than `--update-with-template`, I have added an additional flag `--template`, so you would call `go test -update -template` to trigger the new behavior.

This is just an alternative idea and I'm happy to update with your original suggestion if you prefer that.

Since you have now merged #45, I have merged `master` and added a `GOLDIE_TEMPLATE` env var too.